src/spec.c

```diff
@@ -1571,29 +1583,29 @@ IndexSpec *IndexSpec_CreateFromRdb(RedisModuleCtx *ctx, RedisModuleIO *rdb, int
   CursorList_AddSpec(&RSCursors, sp->name, RSCURSORS_DEFAULT_CAPACITY);
   RedisModule_FreeString(ctx, specKey);
 
-  sp->smap = NULL;
   if (sp->flags & Index_HasSmap) {
     sp->smap = SynonymMap_RdbLoad(rdb, encver);
+    if (sp->smap == NULL)
+      goto cleanup;
   }
   if (IndexSpec_OnCreate) {
     IndexSpec_OnCreate(sp);
   }
 
-  sp->timeout = RedisModule_LoadUnsigned(rdb);
+  sp->timeout = LoadUnsigned_IOError(rdb, goto cleanup);
 
-  size_t narr = RedisModule_LoadUnsigned(rdb);
+  size_t narr = LoadUnsigned_IOError(rdb, goto cleanup);
   for (size_t ii = 0; ii < narr; ++ii) {
     QueryError _status;
     size_t dummy;
-    char *s = RedisModule_LoadStringBuffer(rdb, &dummy);
+    char *s = LoadStringBuffer_IOError(rdb, &dummy, goto cleanup);
     int rc = IndexAlias_Add(s, sp, 0, &_status);
     RedisModule_Free(s);
     RedisModule_Log(NULL, "notice", "Loading existing alias failed");
```

Comment:
Why do we always print this log message?

Reply:
Seems to be done unintentionally by PR #2045

@ashtul We should add a condition to the log, agree?
```
if (rc != REDISMODULE_OK) {
  RedisModule_Log(NULL, "notice", "Loading existing alias failed");
}

```